chia/_tests/pools/test_pool_rpc.py

```diff
@@ -1020,3 +1022,48 @@ async def status_is_leaving_no_blocks() -> bool:
 
         # Eventually, leaves pool
         assert await status_is_farming_to_pool()
+
+    @pytest.mark.anyio
+    async def test_join_pool_twice(
+        self,
+        setup: Setup,
+        self_hostname: str,
+    ) -> None:
+        full_node_api, wallet_node, our_ph, _total_block_rewards, client = setup
+
+        await wallet_node.server.start_client(PeerInfo(self_hostname, full_node_api.server.get_port()), None)
+        await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(our_ph))
+        await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(our_ph))
+        await full_node_api.wait_for_wallet_synced(wallet_node=wallet_node, timeout=20)
+
+        creation_tx: TransactionRecord = await client.create_new_pool_wallet(
+            target_puzzlehash=our_ph,
+            pool_url="https://pool.example.com",
+            relative_lock_height=uint32(10),
+            backup_host="",
+            mode="new",
+            state="FARMING_TO_POOL",
+            fee=uint64(0),
+        )
+        await full_node_api.process_transaction_records(records=[creation_tx])
+        await full_node_api.wait_for_wallet_synced(wallet_node=wallet_node, timeout=20)
+
+        summaries_response = await client.get_wallets(WalletType.POOLING_WALLET)
+        assert len(summaries_response) == 1
+        wallet_id: int = summaries_response[0]["id"]
+
+        async def status_is_farming_to_pool(w_id: int) -> bool:
+            pw_status: PoolWalletInfo = (await client.pw_status(w_id))[0]
+            return pw_status.current.state == PoolSingletonState.FARMING_TO_POOL.value
+
+        await time_out_assert(45, status_is_farming_to_pool, True, wallet_id)
+
+        # Test joining the same pool via the RPC client
+        with pytest.raises(ValueError, match="Already farming to pool"):
```

Comment:
Needs an import as well.

```suggestion
        with pytest.raises(ResponseFailureError, match="Already farming to pool"):
```